engine/src/main/java/io/zeebe/engine/processing/deployment/TransformingDeploymentCreateProcessor.java

```diff
@@ -50,18 +58,38 @@
   private final CatchEventBehavior catchEventBehavior;
   private final KeyGenerator keyGenerator;
   private final ExpressionProcessor expressionProcessor;
+  private final TypedStreamWriterProxy typedStreamWriterProxy;
+  private final StateWriter stateWriter;
+  private final List<Integer> partitions;
+  private final DeploymentDistributionRecord deploymentDistributionRecord;
+  //  private final DeploymentDistributor deploymentDistributor;
 
   public TransformingDeploymentCreateProcessor(
       final ZeebeState zeebeState,
       final CatchEventBehavior catchEventBehavior,
-      final ExpressionProcessor expressionProcessor) {
+      final ExpressionProcessor expressionProcessor,
+      final int partitionsCount) {
+    final EventAppliers eventAppliers = new EventAppliers(zeebeState);
     workflowState = zeebeState.getWorkflowState();
     eventScopeInstanceState = zeebeState.getEventScopeInstanceState();
     timerInstanceState = zeebeState.getTimerState();
     keyGenerator = zeebeState.getKeyGenerator();
-    deploymentTransformer = new DeploymentTransformer(zeebeState, expressionProcessor);
+    typedStreamWriterProxy = new TypedStreamWriterProxy();
+    stateWriter = new StateWriter(typedStreamWriterProxy, eventAppliers);
+    deploymentTransformer = new DeploymentTransformer(stateWriter, zeebeState, expressionProcessor);
     this.catchEventBehavior = catchEventBehavior;
     this.expressionProcessor = expressionProcessor;
+
+    // partitions
+    if (partitionsCount > Protocol.DEPLOYMENT_PARTITION) {
+      partitions =
+          IntStream.range(Protocol.DEPLOYMENT_PARTITION + 1, partitionsCount + 1)
+              .boxed()
+              .collect(Collectors.toList());
+    } else {
+      partitions = List.of();
+    }
```

Comment:
We can simplify the statement because `IntStream` can handle empty ranges. 

```suggestion
    // partitions
    partitions =
        IntStream.range(Protocol.DEPLOYMENT_PARTITION + 1, partitionsCount + 1)
            .boxed()
            .collect(Collectors.toList());
```

(optional) Currently, the code implies that `Protocol.DEPLOYMENT_PARTITION` is the first partition. We could write it more explit like this:

```
partitions = IntStream.range(
    Protocol.START_PARTITION_ID, Protocol.START_PARTITION_ID + partitionsCount)
  .filter(partition -> partition != Protocol.DEPLOYMENT_PARTITION)
  .boxed()
  .collect(Collectors.toList());
```